### PR TITLE
OPERATORS: Finalize legal findings, Update smoosh + exit gw, update --host changes

### LINKED ISSUE #4043: OPERATORS: Look into `--host` problems
### Quick note to self - will look into it later

There are three options:

An existing gateway, and the operator doesn't want to run WSS:
Update the binary and restart the gateway, nothing more is required.
An existing gateway, and the operator wants to run WSS: 
Use the script above
An operator setting up a gateway for the first time:  
New flags will need to be added to the init/run command. The --host is still accepted for now, but can and should be replaced with --listening-address, this is the IP address which is used for receiving sphinx packets and listening to client data. Another flag is required --public-ips which is a comma separated list of ip ip’s that are announced to the nym-api, it is usually the address which is used for bonding. And lastly, if the operator wishes to run WSS, an optional --hostname flag is also required, being something like mainnet-gateway2.nymtech.net
Make sure to enable all necessary ports on the gateway. The gateway will then be accessible on something like: http://85.159.211.99:8080/api/v1/swagger/index.html

are you seeing something like this node attempted to announce an invalid public address: 0.0.0.0. Please modify [host.public_ips] section of your config. ?

If so the fields are going to be slightly different:
--listening-address "0.0.0.0" --public-ips "$(curl -4 https://ifconfig.me)"

### PR DESCRIPTION
**Pull request by serinko**
## Description

Document changes in Smoosh, exit gateways and Swiss legal findings, document #4024 

## Checklist:

- [x] Swiss legal finding
- [x] warning: not a legal advice
- [x]  smoosh updates (possible ways to handle the exit GW only option)
- [x] exit policy update
- [x] resolve #4043

### CODE REVIEW

File: documentation/operators/src/faq/smoosh-faq.md
Comment (tommyv1987) [inline documentation/operators/src/faq/smoosh-faq.md]: This is me being pedantic, but I'm not overly keen on the "how such binary" comment teeth 😬 maybe: 

We are exploring two potential methods for implementing binary functionality in practice and will provide information in advance. The options are: ....

File: documentation/operators/src/faq/smoosh-faq.md
Comment (tommyv1987) [inline documentation/operators/src/faq/smoosh-faq.md]: `Nodes that function as exit gateways`

File: documentation/operators/src/faq/smoosh-faq.md
Comment (tommyv1987) [inline documentation/operators/src/faq/smoosh-faq.md:39]: We require all nodes to include exit gateway functionality?

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: I am glad for such feedback as I am not native speaker (of any language), lol.

**Comment by serinko** [comment]
@tommyv1987 please give it a review, but don't merge it just yet as I'm waiting for legal confirmations.

File: documentation/operators/src/legal/exit-gateway.md
Comment (tommyv1987) [inline documentation/operators/src/legal/exit-gateway.md]: In accordance with the [Tor Null 'deny' list] and [Tor reduced policy] (https://tornull.org/tor-reduced-reduced-exit-policy.php), which are two established safeguards.

File: documentation/operators/src/legal/exit-gateway.md
Comment (tommyv1987) [inline documentation/operators/src/legal/exit-gateway.md]: For exit relays on ports 80 and 443, the Gateways will exhibit an HTML page resembling the one proposed by [Tor] (https://gitlab.torproject.org/tpo/core/tor/-/raw/HEAD/contrib/operator-tools/tor-exit-notice.html). By doing so, the operator will be able to disclose details regarding their Gateway, including the currently configured exit policy, all without the need for direct correspondence with regulatory or law enforcement agencies.

File: documentation/operators/src/legal/exit-gateway.md
Comment (tommyv1987) [inline documentation/operators/src/legal/exit-gateway.md]: The principal change in the new configuration is to make this short allow list more permissive. Nym Tech.Net/.wellknown/network-requester/exit-policy.txt An exit policy will restrict the hosts to which Nym Mixnet and Nym VPN users are permitted to connect.

File: documentation/operators/src/legal/exit-gateway.md
Comment (serinko) [reply documentation/operators/src/legal/exit-gateway.md]: done

File: documentation/operators/src/legal/exit-gateway.md
Comment (serinko) [reply documentation/operators/src/legal/exit-gateway.md]: done

File: documentation/operators/src/legal/exit-gateway.md
Comment (serinko) [reply documentation/operators/src/legal/exit-gateway.md]: done

**Comment by serinko** [comment]
OK, this bloated PR shall be ready for review. Only two points of clarification on my side are:
1. What's the path of /nym repo on the backend runner
2. The part in WSS setup https://github.com/nymtech/nym/blob/operators/smoosh-updates/documentation/operators/src/nodes/maintenance.md#wss-on-a-new-gateway , it's originally from @benedettadavico

File: documentation/operators/src/SUMMARY.md
Comment (mfahampshire) [inline documentation/operators/src/SUMMARY.md]: [[ very minor i know :grimacing:  ]] can we remove the initial `./` here? I think its in one other place in the `SUMMARY` file (need to remove that too) but if we can remove then the style is consistent across the file.

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: *config.toml* should be `config.toml`?

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: ditto with *allowed.list* -> `allowed.list`

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: *allowed.list* should be `allowed.list`?

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: Few small grammatical tweaks: 
* 'Depends which' => 'It depends which' 
* 'will be the final one' => 'will ultimately be used' 
* Remove 'one' after 'first' 
* Re the last line - which gateway functionality specifically? both entry and exit? or is exit also a separate decision?

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: * 'any details' => 'specifics' 
* 'please read' => 'please read the'

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md:85]: Do we have an outline of how this will be monitored?

File: documentation/operators/src/faq/smoosh-faq.md
Comment (mfahampshire) [inline documentation/operators/src/faq/smoosh-faq.md]: Can you elaborate re: 'how many users pay'?

**Review by mfahampshire — CHANGES_REQUESTED** [review]
Minor grammatical stuff mostly otherwise looks great

File: documentation/operators/src/SUMMARY.md
Comment (serinko) [reply documentation/operators/src/SUMMARY.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md]: ✅

File: documentation/operators/src/faq/smoosh-faq.md
Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md:85]: ✅

Comment (serinko) [reply documentation/operators/src/faq/smoosh-faq.md:85]: (Specified that we will measure and share, I don't think we do just yet)

**Comment by serinko** [comment]
I am going through other pages I edited if some of the syntax changes you commented aren't to be done there as well, otherwise good to go.

File: documentation/operators/src/nodes/maintenance.md
Comment (benedettadavico) [inline documentation/operators/src/nodes/maintenance.md:240]: Possibly add that ports 8080 or 80/443 (if they put it through reverse proxy) need to be opened too in order for the gateway swagger page to work

Comment (serinko) [reply documentation/operators/src/nodes/maintenance.md:240]: done

**Comment by serinko** [comment]
Good to go imo

**Review by mfahampshire — APPROVED** [review]

**Review by mfahampshire — APPROVED** [review]

**Comment by serinko** [comment]
Going to read all over the legal part once again for certainty and merge it, thank you.

**Pull request merged**
